.github/workflows/soroban-deploy.yml

```
@@ -0,0 +1,285 @@
+
+      - name: Setup deployment identity
+        run: |
+          echo "${{ secrets.SOROBAN_SECRET_KEY }}" > secret.key
+          soroban config identity add testnet-deployer \
+            --secret-key "$(cat secret.key)"
+          rm secret.key
+
+      - name: Deploy contracts
+        env:
+          CONTRACT_NAME: ${{ github.event.inputs.contract_name }}
+        run: |
```

actionlint (1.7.12)

[error] shellcheck reported issue in this script: SC2086:info:20:45: Double quote to prevent globbing and word splitting

(shellcheck)